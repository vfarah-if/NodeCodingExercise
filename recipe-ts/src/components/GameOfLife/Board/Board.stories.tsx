import { Meta, Story } from '@storybook/react/types-6-0'
import Board, { BoardProps } from './Board'

export default {
  title: 'Component/GameOfLife/Board',
  component: Board,
} as Meta

const BoardTemplate: Story<BoardProps> = args => (
  <>
    <Board {...args}>
      <div>1</div>
      <div>2</div>
      <div>3</div>
      <div>4</div>
    </Board>
  </>
)

export const SimpleBoard = BoardTemplate.bind({})
SimpleBoard.args = { boardSize: 2, cellSize: 40 }
