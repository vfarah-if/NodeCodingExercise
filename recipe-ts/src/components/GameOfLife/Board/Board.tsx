import React from 'react'
import './style/index.css'

export interface BoardProps {
  boardSize?: number
  cellSize?: number
  children?: React.ReactNode
}

const Board: React.FC<BoardProps> = ({
  boardSize = 20,
  cellSize = 50,
  children,
}) => {
  return (
    <div
      className='grid'
      style={{
        display: 'grid',
        gridTemplateColumns: `repeat(${boardSize}, ${cellSize}px)`,
        gridTemplateRows: `repeat(${boardSize}, ${cellSize}px)`,
      }}
    >
      {children}
    </div>
  )
}

export default Board
