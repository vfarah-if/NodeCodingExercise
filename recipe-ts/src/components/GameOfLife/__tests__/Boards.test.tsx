import { render } from '@testing-library/react';
import { SimpleBoard } from '../Board/Board.stories';

describe('Board', () => {
  test('should output a board of simpl items as children', () => {
    const { container } = render(<SimpleBoard {...SimpleBoard.args} />);

    expect(container).toMatchSnapshot();
  });
});
