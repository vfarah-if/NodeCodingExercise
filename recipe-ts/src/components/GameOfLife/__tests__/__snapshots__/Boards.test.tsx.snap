// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board should output a board of simpl items as children 1`] = `
<div>
  <div
    class="grid"
    style="display: grid; grid-template-columns: repeat(2, 40px); grid-template-rows: repeat(2, 40px);"
  >
    <div>
      1
    </div>
    <div>
      2
    </div>
    <div>
      3
    </div>
    <div>
      4
    </div>
  </div>
</div>
`;
